packages/protocol/src/routers/ConnextHandler.sol

```diff
@@ -114,8 +131,10 @@ contract ConnextHandler {
     external
     onlyConnextRouter
   {
-    _failedTxns[transferId] =
-      FailedTxn(transferId, amount, asset, originSender, originDomain, actions, args);
+    if (!isTransferIdRecorded(transferId)) {
```

Comment:
Doing this solves overriding vector, but adds a different attacking vector 
now anyone can pre-call xBundle with the given `transferId` and make the legit `recordFail` revert

we had mentioned this previously as well
https://www.notion.so/0xmacro/Leads-Questions-on-Router-14d2593767a640258c902031573da7c8?pvs=4#c508aacbca3847aca17a1372879ed074
![image](https://github.com/Fujicracy/fuji-v2/assets/46760063/8c1ff913-3c59-4366-9590-30bcab2d4ac5)

Reply:
Hey @abhishekvispute good point, decided to essentially record all attempted transferIds with an increasing nonce. 
Please refer to #536.